components/speedreader/resources/panel/components/main-panel/index.tsx

```diff
@@ -3,40 +3,93 @@ import * as React from 'react'
 import * as S from './style'
 import Toggle from '$web-components/toggle'
 import { FontStyleList, ContentList } from '../lists'
-import ZoomControl from '../zoom-control'
+import FontSizeControl from '../font-size-control'
 import ThemeControl from '../theme-control'
+import { SiteSettings, Theme, FontSize, ContentStyle, FontFamily } from '../../api/browser'
+import BackgroundSVG from '../../svg/background'
+import { getLocale, splitStringForTag } from '$web-common/locale'
+
+interface MainPanelProps {
+  siteSettings: SiteSettings
+  onThemeChange: (theme: Theme) => void
+  onFontSizeChange: (fontSize: FontSize) => void
+  onContentStyleChange: (contentStyle: ContentStyle) => void
+  onFontFamilyChange: (fontFamily: FontFamily) => void
+  onToggleChange: (isOn: boolean) => void
+}
+
+function MainPanel (props: MainPanelProps) {
+  const braveSpeedreaderOffText = splitStringForTag('braveSpeedreaderTurnOffDesc')
+
+  let headerTextElement = null
+  let mainContentElement = (
```

Comment:
How about only creating these nodes if `!props.siteSettings.isEnabled`? If you're having trouble conditionalising them, you could always extract to something like `HeaderContent` and `MainContent` components which return early if `isEnabled` is `false`?

Reply:
What is the issue with how it is now? 

Extracting to its own component means I'd have to prop drill to read values other than isEnabled boolean.
Is your concern around `headerTextElement` being null?

Reply:
The issue is that component instances are being created but not used. It's bad practice IMO and harms readability.

If you're worried about making another component and having overhead then you could at least ensure you're only creating those instances if the site is not enabled:

```ts
let headerTextElement: JSX.Element | null
let mainContentElement: JSX.Element

if (!props.siteSettings.isEnabled) {
    const braveSpeedreaderOffText = splitStringForTag(getLocale('braveSpeedreaderTurnOffDesc'))
    headerTextElement = null
    mainContentElement = (
    <>
        <S.Section>
          {braveSpeedreaderOffText.beforeTag}
          <a href="brave://settings/">{braveSpeedreaderOffText.duringTag}</a>
          {braveSpeedreaderOffText.afterTag}
        </S.Section>
        <S.Bg><BackgroundSVG /></S.Bg>
      </>
    )
    )
}
```

You could also perhaps avoid the overhead by either using React's Context, or by just passing all the props:
```ts
function Header(props: MainPanelProps) {
    
}

function MainContent(props: MainPanelProps) {
    
}

function MainPanel(props: MainPanelProps) {
  return (
    <S.Box>
      <S.HeaderBox>
        <S.HeaderContent>
          <div>{getLocale('braveSpeedreader')}</div>
          <div>
            <Toggle
              brand="shields"
              isOn={props.siteSettings.isEnabled}
              onChange={props.onToggleChange}
            />
          </div>
        </S.HeaderContent>
        <Header {...props} />
      </S.HeaderBox>
      <MainContent {...props} />
    </S.Box>
  )
}
```

Reply:
The default state is always to show as if the Speedreader is turned off. Why do we need the following condition?  `if (!props.siteSettings.isEnabled)`

Adding the above condition will create a code path that will never be reached and TS complains the variables used in the return function is used before assigned.

> Variable 'headerTextElement' is used before being assigned.
> Variable 'mainContentElement' is used before being assigned.

If the goal is to create a new execution context then we can do that with just curly braces. The TS error goes away and we always know the default state without the condition being in the way.

```js
let headerTextElement: JSX.Element | null
let mainContentElement: JSX.Element

  {
    const braveSpeedreaderOffText = splitStringForTag(getLocale('braveSpeedreaderTurnOffDesc'))
    headerTextElement = null
    mainContentElement = (
      <>
          <S.Section>
            {braveSpeedreaderOffText.beforeTag}
            <a href="brave://settings/">{braveSpeedreaderOffText.duringTag}</a>
            {braveSpeedreaderOffText.afterTag}
          </S.Section>
          <S.Bg><BackgroundSVG /></S.Bg>
        </>
    )
  }
```

Reply:
Done. I have cleaned this up and extracted to it's own components.